Merge Xero freeze branch #2

lakchote opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Same logic as https://github.com/Expensify/App/pull/42043
### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$
Related to https://github.com/Expensify/Expensify/issues/349326

Since it's a feature branch that was created during the merge freeze, no specific issue had to be created.
Every PR merged here has been individually reviewed both by an internal engineer and a C+. Also every PR was linked to an issue.


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console
Same logic as https://github.com/Expensify/App/pull/42043

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
NA

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console
Same as in tests 
### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I verified that similar component doesn't exist in the codebase
- [x] I verified that all props are defined accurately and each prop has a `/** comment above it */`
- [x] I verified that each file is named correctly
- [x] I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
- [x] I verified that the only data being stored in component state is data necessary for rendering and nothing else
- [x] In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
- [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
- [x] I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
- [x] I verified that all JSX used for rendering exists in the render method
- [x] I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

rushatgabhane committed:
add routes

rushatgabhane committed:
add screen map

rushatgabhane committed:
add navigation map

rushatgabhane committed:
link to bill status page and add default page

rushatgabhane committed:
add selection screen

ShridharGoel committed:
Add sync functionality in Xero connection

rushatgabhane committed:
add selection screen

ShridharGoel committed:
Update

ShridharGoel committed:
Update PolicyAccountingPage.tsx

ShridharGoel committed:
Update PolicyAccountingPage.tsx

ShridharGoel committed:
Add new command and JSdoc

ShridharGoel committed:
Merge branch '41320' of https://github.com/ShridharGoel/ExpensifyApp into 41320

ShridharGoel committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into 41320

ShridharGoel committed:
Update

rushatgabhane committed:
fix merge

rushatgabhane committed:
unlink from advanced

rushatgabhane committed:
link export page

rushatgabhane committed:
fix merge

rushatgabhane committed:
update route to export

rushatgabhane committed:
add lang

rushatgabhane committed:
fix nav

rushatgabhane committed:
use updated routes and lang

rushatgabhane committed:
update status name

rushatgabhane committed:
change order

rushatgabhane committed:
change order

rushatgabhane committed:
handle undefined

rushatgabhane committed:
Merge branch 'Expensify:main' into bill-status

rushatgabhane committed:
Update src/pages/workspace/accounting/xero/export/XeroPurchaseBillStatusSelectorPage.tsx

Co-authored-by: Manan <manan.jadhav@gmail.com>

rushatgabhane committed:
Update src/SCREENS.ts

Co-authored-by: Manan <manan.jadhav@gmail.com>

SzymczakJ committed:
add offline support for XeroOrganizationConfiguration

SzymczakJ committed:
Merge branch 'main' into @szymczak/handle-offline-state-xero-organization

SzymczakJ committed:
fix PR comments

SzymczakJ committed:
fix XeroOrganization bug

SzymczakJ committed:
use ConnectionLayout in XeroOrganizationConfigurationPage

SzymczakJ committed:
Merge branch 'main' into @szymczak/handle-offline-state-xero-organization

rushatgabhane committed:
Merge branch 'main' of github.com:rushatgabhane/exfy into bill-status

SzymczakJ committed:
fix padding on XeroOrganizationList

SzymczakJ committed:
fix safe area insets

ShridharGoel committed:
Update

ShridharGoel committed:
Updates

SzymczakJ committed:
Merge branch 'main' into @szymczak/handle-offline-state-xero-organization

SzymczakJ committed:
fix typescript errors

SzymczakJ committed:
make taxes read only

rushatgabhane committed:
Update src/languages/es.ts

Co-authored-by: Rocio Perez-Cano <pecanoro@users.noreply.github.com>

rushatgabhane committed:
Update src/languages/es.ts

Co-authored-by: Rocio Perez-Cano <pecanoro@users.noreply.github.com>

rushatgabhane committed:
Update src/languages/es.ts

Co-authored-by: Rocio Perez-Cano <pecanoro@users.noreply.github.com>

SzymczakJ committed:
add error message in case of sync failure

ShridharGoel committed:
Merge branch 'main' into 41320

ShridharGoel committed:
Fix lint

mananjadhav committed:
fix: remove slash from url

mananjadhav committed:
fix: route path

lakchote committed:
Merge pull request #41562 from ShridharGoel/41320

Add sync functionality in Xero connection

lakchote committed:
Merge pull request #41983 from software-mansion-labs/@szymczak/handle-offline-state-xero-organization

Add proper offline state handling for XeroOrganizationConfigurationPage

lakchote committed:
Merge pull request #42324 from mananjadhav/fix-xero-customers-console-error

fix: console error on xero customers page.

lakchote committed:
Merge pull request #41559 from rushatgabhane/bill-status

[Wave Collect][Xero][Advanced] Bill status selector page

mananjadhav committed:
fix: change selected text

mananjadhav committed:
fix: added padding

mananjadhav committed:
fix: default selection

rushatgabhane committed:
hide policy guides as exporters

mananjadhav committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into xero-tracking-categories-fixes

mananjadhav committed:
fix: select first item in list

mananjadhav committed:
fix: default select first item

mananjadhav committed:
fix: default select first item

tienifr committed:
Console error shows up when opening Preferred exporter

lakchote committed:
Merge pull request #42347 from mananjadhav/xero-tracking-categories-fixes

[Wave Collect][Xero] Tracking categories fixes

SzymczakJ committed:
Merge branch 'main' into @szymczak/error-message-sync-failure

SzymczakJ committed:
add translations

SzymczakJ committed:
delete unrelated changes

SzymczakJ committed:
Merge branch 'main' into @szymczak/make-taxes-read-only-when-connected

SzymczakJ committed:
fix PR comments

SzymczakJ committed:
Merge remote-tracking branch 'expensify-real/xero-merge-freeze' into @szymczak/error-message-sync-failure

SzymczakJ committed:
Merge remote-tracking branch 'expensify-real/xero-merge-freeze' into @szymczak/make-taxes-read-only-when-connected

hungvu193 committed:
Fix: QBO briefly shows when syncing Xero

hungvu193 committed:
fix prettier

hungvu193 committed:
update Spanish translation

lakchote committed:
Merge pull request #42346 from rushatgabhane/xero-exporter

[Wave Collect][Xero] Hide guides as exporters

lakchote committed:
Merge pull request #42372 from tienifr/fix/42195

Console error shows up when opening Preferred exporter

SzymczakJ committed:
fix connections lastSync unsafe access

mananjadhav committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into xero-default-select-options

Commit:
Merge pull request #42149 from s77rt/multi-level-tags-fix-not-found

[NoQA] Add orderWeight to tag view and tag edit routes

lakchote committed:
Merge pull request #42192 from software-mansion-labs/@szymczak/make-taxes-read-only-when-connected

Make taxes read-only when connected to accounting integration

lakchote committed:
Merge pull request #42383 from hungvu193/fix-42376

[Wave Collect] [Xero]  Fix: When you click “Sync now” you see the Quickbooks Online text very briefly

mananjadhav committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into xero-default-select-options

mananjadhav committed:
refactor: move bank selection to utils

mananjadhav committed:
refactor: replace with utils

mananjadhav committed:
refactor: replace with utils

mananjadhav committed:
refactor: lint fixes

lakchote committed:
Merge pull request #42360 from mananjadhav/xero-default-select-options

[Wave Collect] [Xero] Default select options

SzymczakJ committed:
fix taxes not blocking on more features page

hungvu193 committed:
Update category integration name

trjExpensify commented:
Running a fresh build on this now, related to: https://expensify.slack.com/archives/C036QM0SLJK/p1716374148109909?thread_ts=1716238514.357639&cid=C036QM0SLJK

lakchote committed:
disable taxes if previously enabled on disconnect

rushatgabhane committed:
rm header overflow menu

rushatgabhane committed:
rm header overflow menu

rushatgabhane committed:
hide 3dot

rushatgabhane committed:
hide 3dot

rushatgabhane committed:
undo all beta

lakchote committed:
Merge pull request #42462 from hungvu193/fix-42458

[Wave Collect] [Xero] Make the integration name in Categories dynamic

lakchote committed:
change logic

lakchote committed:
remove taxes in lhn when connecting to QBO

mananjadhav committed:
fix: replace connection name

mananjadhav committed:
feat: added content for taxes page

mananjadhav committed:
fix: update translate path

mananjadhav committed:
feat: footer text for more features

mananjadhav committed:
feat: added content for more features

mananjadhav committed:
fix: add margin

mananjadhav committed:
refactor: prettier fix

mananjadhav committed:
fix: update styling

SzymczakJ committed:
fix PR comments

lakchote committed:
Merge pull request #42461 from software-mansion-labs/@szymczak/fix-taxes-not-blocked-with-integration

Fix taxes not blocked with accounting integration connected

lakchote committed:
Merge pull request #42468 from rushatgabhane/xero-rm-3-dot

[Xero] Remove overflow menu from accounting header

techievivek committed:
Merge pull request #42467 from Expensify/lucien/xero-fix-disconnect-taxes

[Wave Collect] [Xero] Disable taxes if previously enabled when connecting to QBO

SzymczakJ committed:
change translations

trjExpensify commented:
Wwww... cheeky conflict here now. 🤔

arosiclair committed:
Merge pull request #42315 from Expensify/hayata-call-api-for-multi-level-tags

Enable / Disable multi level tags

lakchote committed:
make translations dynamic for xero tracking categories

lakchote committed:
use only one route for tracking category mapping

lakchote committed:
use one screen for the tracking category mapping

lakchote committed:
handle navigation for the new page for tracking category

lakchote committed:
add new types for tracking category mapping

lakchote committed:
map the tracking category page to rhp

lakchote committed:
create function to get tracking categories

lakchote committed:
modify component for dynamic translations

lakchote committed:
make the handling of tracking categories dynamic

lakchote committed:
fix style

lakchote committed:
delete specific tracking categories pages

lakchote committed:
make the handling of tracking categories dynamic

lakchote committed:
Merge branch 'main' into xero-merge-freeze

mananjadhav committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into xero-educational-messages

mananjadhav committed:
fix: revert in-line text

mananjadhav committed:
fix: revert in-line text

mananjadhav committed:
fix: added period in all content

mananjadhav committed:
chore: resolve merge issues

mananjadhav committed:
chore: resolve merge issues

mananjadhav committed:
fix: use var

mananjadhav committed:
chore: resolve merge issues

mananjadhav committed:
style: lint fixes

mananjadhav committed:
feat: added shouldBeBlocked to ConnectionLayout

mananjadhav committed:
fix: added checks for connection import pages

mananjadhav committed:
fix: added block check for xero pages

mananjadhav committed:
style: lint fixes

lakchote committed:
fix lint

mananjadhav committed:
refactor: added connection name

mananjadhav committed:
refactor: use connection name

mananjadhav committed:
refactor: use connection name

mananjadhav committed:
refactor: use connection layout for import page

mananjadhav committed:
style: lint fixes

mananjadhav committed:
fix: added connection name to pending pages

lakchote committed:
Merge pull request #42619 from mananjadhav/xero-access-block

[Wave collect] [Xero] Block Xero screens if Xero is not connected.

lakchote committed:
Merge branch 'xero-merge-freeze' into lucien/fix-xero-tracking-categories

lakchote committed:
fix lint

lakchote committed:
fix typo

SzymczakJ committed:
Merge branch 'xero-merge-freeze' into @szymczak/error-message-sync-failure

lakchote committed:
encode category name

lakchote committed:
decode category name

SzymczakJ committed:
fix style

SzymczakJ committed:
fix styles

lakchote committed:
fix isSelected

lakchote committed:
use already decoded category name

lakchote committed:
fix lint

SzymczakJ committed:
fix lint errors

SzymczakJ committed:
fix type error

mananjadhav committed:
feat: add lock icon for switch

mananjadhav committed:
feat: state modal

mananjadhav committed:
feat: added features modal

mananjadhav committed:
refactor: modal state

mananjadhav committed:
refactor: language format

mananjadhav committed:
refactor: language and connection name

mananjadhav committed:
style: prettier fixes

lakchote committed:
Merge pull request #42593 from Expensify/lucien/fix-xero-tracking-categories

Fix tracking categories not dynamic

SzymczakJ committed:
fix PR comments

mananjadhav committed:
Merge branch 'xero-merge-freeze' of https://github.com/Expensify/App into xero-more-features-text

mananjadhav committed:
feat: added es translations

mananjadhav committed:
refactor: remove title

mananjadhav committed:
fix: added dependency

mananjadhav committed:
style: lint fixes

mananjadhav committed:
fix: update copy

mananjadhav committed:
refactor: remove unused var

mananjadhav committed:
fix: spanish copy

mananjadhav committed:
style: lint fixes

mananjadhav committed:
fix: spanish copy

mananjadhav committed:
fix: captilize

lakchote committed:
Merge pull request #42662 from mananjadhav/xero-more-features-text

[Wave Collect] [Xero] More features text

lakchote committed:
Merge pull request #42307 from software-mansion-labs/@szymczak/error-message-sync-failure

Add error message in case of integration sync failure

lakchote committed:
Merge branch 'main' into xero-merge-freeze

lakchote committed:
fix lint

mananjadhav committed:
fix: remove unused translations

mananjadhav committed:
fix: header for smallscreen width

mananjadhav committed:
fix: show text in empty state

mananjadhav committed:
fix: boolean check

mananjadhav committed:
refactor: simplify condition

lakchote committed:
Merge pull request #42487 from mananjadhav/xero-educational-messages

[Wave Collect] [Xero/QBO] Update educational messages

lakchote committed:
Merge branch 'main' into xero-merge-freeze

lakchote committed:
fix lint

hungvu193 reviewed (commented):
Looks good. Just one minor change left

lakchote committed:
fix backtick

lakchote commented on src/languages/en.ts:
```diff
@@ -2186,6 +2186,12 @@ export default {
                 subtitle: 'Sync your chart of accounts and more.',
             },
+            connectionsWarningModal: {
+                featureEnabledTitle: `Not so fast...`,
```
Fixed!

hungvu193 commented:
@lakchote Hey can you take a look? I can't seem to connect to Xero or QBO

https://github.com/Expensify/App/assets/16502320/e71b2b9e-8fc3-4a60-a482-904f8db3405c

hungvu193 commented:
From OldDot, it's showing I'm connecting to Xero

https://github.com/Expensify/App/assets/16502320/1dddb851-d6b3-4af2-8d10-f6131beba066

lakchote commented:
> @lakchote Hey can you take a look? I can't seem to connect to Xero or QBO
> 

I'm able to connect just fine (though the error text shows briefly while syncing - this needs to be fixed cc @SzymczakJ).

https://github.com/Expensify/App/assets/21549771/3e7b24c1-55ac-4f43-8816-1d0938f2c59f

Make sure you have an up to date auth token in OldDot, and let me know if that works.

hungvu193 commented:
Ok let me try again

lakchote commented:
Also sometimes the pusher update seems to get stuck which seems like to be your case, can you refresh the page in NewDot? It should show you as synced now.

hungvu193 commented:
Yeah, I think the problem is with Pusher event? I was out for 5 minutes and when I'm back I saw I connected to Xero on NewDot.

hungvu193 commented:
Cool. I will continue with testing

hungvu193 commented:
Is is expected that we will sync again after I changed some options?

https://github.com/Expensify/App/assets/16502320/a4509474-1d6b-494b-b9ec-beb31ec7c8fd

hungvu193 commented:
Also can you please check all the checkmarks of your author checklist?

trjExpensify commented:
>Is is expected that we will sync again after I changed some options?

Yes!

hungvu193 commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16502320/e0a9abb9-d375-41a4-8eaa-bcb34194fb0a


</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16502320/4b40d5d7-4819-4fec-9d3c-e3f3ad56f1c4


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16502320/b970534f-4cbd-4aab-a00c-c9580bc11ae1


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16502320/abf4a14b-0b8e-4636-ad02-33f1f62e2ee6


</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/16502320/ea81ce48-87e5-4d9a-8f98-90d2292ea70e


</details>

Julesssss commented:
> Same logic as https://github.com/Expensify/App/pull/42043

@lakchote have these changes been individually reviewed? Also is there an issue we can point to?

lakchote commented:
> > Same logic as #42043
> 
> @lakchote have these changes been individually reviewed? Also is there an issue we can point to?

Yes, every PR here has been individually reviewed both by an internal engineer and a C+.

There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.

> Is is expected that we will sync again after I changed some options?
> 

Yes cc @trjExpensify

lakchote commented:
> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
 
Yes @Julesssss, every PR here has been individually reviewed both by an internal engineer and a C+.

There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.

> 
> > Is is expected that we will sync again after I changed some options?

Yes, cc @trjExpensify

Julesssss commented:
Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

lakchote commented:
> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR here has been individually reviewed both by an internal engineer and a C+. And they are all tied to one issue.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR merged here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR merged here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

Done! I've also added more context.

lakchote commented:
> Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

Done! I've also added more context.

Julesssss commented:
Thanks. It's looking good to me, ready for you to merge. Perhaps we should hold on the site issues that started 5 mins ago but I'll leave that up to you

lakchote commented:
> Thanks. It's looking good to me, ready for you to merge. Perhaps we should hold on the site issues that started 5 mins ago but I'll leave that up to you

Thank you! I'd say we can merge it, given the fact that it's behind two betas (`Accounting` and `Xero` beta) and as no real user currently is able to interact with it. But before that there's conflicts that need resolution again 😄

lakchote committed:
Merge branch 'main' into xero-merge-freeze

lakchote commented:
@Julesssss could you please review it again? Then I'll merge it for reasons [here](https://github.com/Expensify/App/pull/42340#issuecomment-2139550859), and also given the fact conflicts happen often.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9309047748) to staging by https://github.com/lakchote in version: 1.4.77-11 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9309047748) to staging by https://github.com/lakchote in version: 1.4.77-11 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9368314546) to production by https://github.com/mountiny in version: 1.4.78-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
